src/content/en/tools/workbox/guides/advanced-recipes.md

```diff
@@ -53,32 +63,40 @@ if ('serviceWorker' in navigator) {
 ```
 
 This code uses the [`workbox-window`](/web/tools/workbox/modules/workbox-window)
-package to register a service worker and respond if it gets stuck in the waiting
+package to register a service worker and react if it gets stuck in the waiting
 phase.
 
 When a waiting service worker is found we inform the user that an updated
 version of the site is available and prompt them to reload. If they accept the
 prompt, we `postMessage()` the new service worker telling it to run
-`skipWaiting()` meaning it'll start to activate.
+`skipWaiting()`, meaning it'll start to activate. Once the new service worker
+has activated and taken control, we reload the current page, causing the latest
+version of all the precached assets to be displayed.
 
 Note: This is one possible approach. For a more in-depth explanation of the
 problem as well as alternative approaches, see this
 [article by Redfin Engineering](https://redfin.engineering/how-to-fix-the-refresh-button-when-using-service-workers-a8e27af6df68).
 
-**Add to your service worker**
+**Code in your service worker**
+
+If you're using one of the Workbox build tools in `GenerateSW` mode, and you
+have `skipWaiting` set to `false` (the default), then this code will
+automatically be included in your generated service worker file, and you don't
+need to add it yourself.
+
+If you're writing your own service worker code, perhaps in conjunction with a
+Workbox build tool in `InjectManifest` mode, then you need to add this to your
+service worker file yourself:
 
 ```javascript
 addEventListener('message', (event) => {
   if (event.data && event.data.type === 'SKIP_WAITING') {
     skipWaiting();
-
-    // Reply to the window so it knows `skipWaiting()` was called.
-    event.ports[0].postMessage(true);
```

Comment:
Why did you remove this?

Reply:
The way the code is written now, there's nothing that executes after the call to `wb.messageSW({type: 'SKIP_WAITING'})`, so there's no reason to wait for a response.

Reply:
Previously I was using `await wb.messageSW()`, which would await a reply and ensure skipWaiting had been called before executing the next line of code.

Reply:
The other issue is that the `message` handler in the service worker you get when using `GenerateSW` with `skipWaiting: false` (courtesy of https://github.com/GoogleChrome/workbox/pull/1929) doesn't call `event.ports[0].postMessage(true)`.

Reply:
Hmmm, do you think we should change that. We could maybe even do a 4.1.1 release?

I don't mind code that waits for controlling, but my preference would be to always reply to the message event if we can. Reason being that in the future we might have to implement timeouts for messages and I'd rather not have to wait for a timeout in any cases other than version mis-matches.

Reply:
"reply to the message event **if we can**" is relevant here, since our `message` event listeners can't assume that they'll always be passed a `MessageChannel`. I know that `workbox-window` will always pass in one, but there is nothing stopping folks from calling `postMessage()` without using `workbox-window`.

So we'd need to conditionally check to see if `event.ports[0]` existed.

I am cool with that, but rather than rushing this out in a v4.1.1, why not implement that sort of standardization in `workbox-messages` and then we can build on top of that in a future release?

Reply:
We can do that, however, if we're not `await`-ing the SW response, then we probably do need to wait until the controller change happens.

Reply:
Yup.

In other words, we can keep that `'controlling'` event listener logic that's already in this PR?

Reply:
SGTM